pkg/endpoint/endpoint.go

```diff
@@ -266,37 +266,42 @@ func RunK8sCiliumEndpointSyncGC() {
 		scopedLog.WithError(err).Error("Not starting controller because unable to get cilium k8s client")
 		return
 	}
+	k8sClient := k8s.Client()
 
 	// this dummy manager is needed only to add this controller to the global list
 	controller.NewManager().UpdateController(controllerName,
 		controller.ControllerParams{
 			RunInterval: 1 * time.Minute,
 			DoFunc: func() error {
-				nodes := node.GetNodes()
-
 				// Don't run if there are no other known nodes
 				// Only run with a probability of 1/(number of nodes in cluster). This
 				// is because this controller runs on every node on the same interval
 				// but only one is neede to run.
+				nodes := node.GetNodes()
 				if len(nodes) <= 1 || runThrottler.Int63n(int64(len(nodes))) != 0 {
 					return nil
 				}
 
+				clusterPods, err := k8sClient.CoreV1().Pods("").List(meta_v1.ListOptions{})
+				if err != nil {
+					scopedLog.WithError(err).Error("Unable to get list of pods to GC CEP")
```

Comment:
Shouldn't here return an error?

Reply:
yes. I remember making this a log but I have no idea why!